src/SerilogTracing/TracingConfiguration.cs

```diff
@@ -69,47 +78,85 @@ public IDisposable TraceToSharedLogger()
 
     IDisposable EnableTracing(Func<ILogger> logger)
     {
-        var instrumentors = Instrument.GetInstrumentors().ToArray();
-        var activityListener = new ActivityListener();
-        var diagnosticListenerSubscription = DiagnosticListener.AllListeners.Subscribe(new DiagnosticListenerObserver(instrumentors));
-        var disposeProxy = new DisposeProxy(diagnosticListenerSubscription, activityListener);
-        
         ILogger GetLogger(string name)
         {
             var instance = logger();
-            return !string.IsNullOrWhiteSpace(name) ? instance.ForContext(Constants.SourceContextPropertyName, name) : instance;
+            return !string.IsNullOrWhiteSpace(name)
+                ? instance.ForContext(Constants.SourceContextPropertyName, name)
+                : instance;
         }
 
-        Sample.ConfigureSampling(activityListener);
-        
-        // Note, this will not be reevaluated if the minimum level dynamically changes.
-        activityListener.ShouldListenTo = source => source.Name == Core.Constants.SerilogActivitySourceName || GetLogger(source.Name).IsEnabled(LogEventLevel.Fatal);
+        var activityListener = new ActivityListener();
+        var disposeProxy = new DisposeProxy(activityListener,
+            DiagnosticListener.AllListeners.Subscribe(
+                new DiagnosticListenerObserver(Instrument.GetInstrumentors().ToArray())));
+
+        var levelMap = InitialLevel.GetOverrideMap();
+
+        // NOTE: We may want a config setting to only listen to specific sources
```

Comment:
_We may want an opt-in to performing level checks eagerly here. It would be a performance win, but would also prevent dynamic log level changes from being effective._